### ISSUE TITLE
Redundant Optional in Pyreverse

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#9014]
### Bug description

```python
class A:
    q: int | None = None
```

Pyreverse generates:

```mmd
classDiagram
  class A {
    q : Optional[int | None]
  }
```

`Optional[int | None]` is redundant.  It should just be `int | None`.

### Command used

```shell
pyreverse -o mmd
```

### Pylint output

```shell
classDiagram
  class A {
    q : Optional[int | None]
  }
```

### Expected behavior

classDiagram
  class A {
    q : int | None
  }

### Pylint version

```shell
pylint 3.0.0a8-dev0
astroid 3.0.0a10-dev0
```
